src/JsonModelGenerator/Program.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace JsonModelGenerator
+{
+    class Program
+    {
+        private static readonly IReadOnlyDictionary<string, string> BuiltInTypes = new Dictionary<string, string>(StringComparer.Ordinal)
+        {
+            { "int", "Int" },
+            { "string", "String" },
+            { "Guid", "Guid" },
+            { "DateTime?", "NullableDateTime" },
+            { "bool", "Bool" }
+        };
+
+        public static string GenerateCode(string sourceFilePath, string @namespace)
+        {
+            string className = Path.GetFileNameWithoutExtension(sourceFilePath);
+            string[] lines = File.ReadAllLines(sourceFilePath);
+            HashSet<string> usings = new HashSet<string>(StringComparer.Ordinal);
+            List<string> properties = new List<string>();
+            List<string> builderLines = new List<string>();
+
+            for (int i = 0; i < lines.Length; ++i)
+            {
+                if (string.IsNullOrWhiteSpace(lines[i]))
+                {
+                    continue;
+                }
+
+                if (lines[i][0] == '#')
+                {
+                    usings.Add($"using {lines[i].Substring(1)};");
+                }
+                else if (lines[i][0] == '@')
+                {
+                    continue;
+                }
+                else
+                {
+                    lines[i] = lines[i].Trim();
+                    int whitespaceIndex = lines[i].IndexOfAny(" \t".ToCharArray());
+                    string propertyName = lines[i].Substring(0, whitespaceIndex);
+                    string compactifiedType = lines[i].Substring(whitespaceIndex).Trim();
+
+                    if (compactifiedType.EndsWith("{}", StringComparison.Ordinal) || compactifiedType.EndsWith("[]", StringComparison.Ordinal))
+                    {
+                        usings.Add("using System.Collections.Generic;");
+                        (string property, string builderLine) = GeneratePropertyForArrayOrDictionary(className, compactifiedType, propertyName, usings);
+                        properties.Add(property);
+                        builderLines.Add(builderLine);
+                    }
+                    else
+                    {
+                        if (Regex.IsMatch(compactifiedType, @"(Guid|DateTime)(\?)?(?=[$\[\{])"))
```

Comment:
Should this regex be anchored? I'm not sure of the full syntax of the file.

Nit: in the character class, the `[` and `{` don't need to be escaped, that part could be `[$[{]` - seems a lot more readable.

Also, might want to consider storing this regex in a static, as a pre-compiled regex. Since the program is relatively small (has few regex's), it probably stays stored in the implicit cache, so may not be relevant to pre-compile.